test/types.ts

```diff
@@ -1,9 +1,13 @@
+interface ExtendedArray<T> extends Array<T> {}
+
 export type ComplexNestedRequired = {
   simple: number;
   nested: {
     date: Date;
     func: () => string;
     array: { bar: number }[];
+    readonlyArray: ReadonlyArray<{ bar: number }>;
+    extendedArray: ExtendedArray<{ bar: number }>;
```

Comment:
the way those types are created in this file won't allow you to add an extended arrays so I suggest removing it

The reason is that all the complex nested types here are the results of calling `DeepRequired` therefore extended array type will be converted to Array type